addons/sale/models/sale.py

```diff
@@ -476,31 +476,23 @@ def create(self, vals):
         result = super(SaleOrder, self).create(vals)
         return result
 
-    def _write(self, values):
-        """ Override of private write method in order to generate activities
-        based in the invoice status. As the invoice status is a computed field
-        triggered notably when its lines and linked invoice status changes the
-        flow does not necessarily goes through write if the action was not done
-        on the SO itself. We hence override the _write to catch the computation
-        of invoice_status field. """
-        if self.env.context.get('mail_activity_automation_skip'):
```

Comment:
you've lost the skip based on the context key here

Reply:
tru, put it back, though it's really just an optimisation (both `activity_unlink` and `activity_schedule` check for it).